src/Data/Color/ColorFactory.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/* Copyright (c) 2017 Nils Haagen <nils.haagen@concepts-and-training.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Data\Color;
+/**
+ * Builds a Color from either hex- or rgb values.
+ *
+ * @author Nils Haagen <nils.haagen@concepts-and-training.de>
+ */
+class ColorFactory {
+
+	/**
+	 * Determine type of input and validate it, then build a color.
+	 * A Color can be constructed with an array of rgb-integers or from
+	 * a hex-value both short and longhand notation.
+	 *
+	 * @param  string|int[] $value
+	 * @throws \InvalidArgumentException
+	 * @return Color
+	 */
+	public function build($value) {
+
+		if(is_array($value)) {
+			$this->validateRGB($value);
```

Comment:
Please change names from "validate*" to "check*". This is a convention we also use in the UI-framework and I try to use in general.